// react-materialize/lib/Checkbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _idgen = _interopRequireDefault(require("./idgen"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var Checkbox = function Checkbox(_ref) {
  var checked = _ref.checked,
      onChange = _ref.onChange,
      filledIn = _ref.filledIn,
      label = _ref.label,
      disabled = _ref.disabled,
      value = _ref.value,
      id = _ref.id,
      indeterminate = _ref.indeterminate;
  var computedId;

  if (indeterminate) {
    computedId = 'indeterminate-checkbox';
  } else if (id) {
    computedId = id;
  } else {
    computedId = (0, _idgen.default)();
  }

  return _react.default.createElement("label", {
    htmlFor: computedId
  }, _react.default.createElement("input", {
    id: computedId,
    className: (0, _classnames.default)({
      'filled-in': filledIn
    }),
    disabled: disabled,
    onChange: onChange,
    type: "checkbox",
    defaultChecked: checked,
    value: value
  }), _react.default.createElement("span", null, label));
};

Checkbox.propTypes = {
  /*
   * checkbox value
   */
  value: _propTypes.default.string.isRequired,

  /*
   * filled-in styled checkbox
   */
  filledIn: _propTypes.default.bool,

  /*
   * label next to checkbox
   */
  label: _propTypes.default.string.isRequired,

  /*
   * Indeterminate Style
   */
  indeterminate: _propTypes.default.bool,

  /*
   * onChange callback
   */
  onChange: _propTypes.default.func,

  /*
   * override id
   * @default idgen()
   */
  id: _propTypes.default.string,

  /*
   * disabled input
   */
  disabled: _propTypes.default.bool,

  /*
   * initialise checkbox checked
   */
  checked: _propTypes.default.bool
};
var _default = Checkbox;
exports.default = _default;
